include/verilated_vpi.cpp

```diff
@@ -1328,281 +1359,294 @@ void vpi_put_delays(vpiHandle /*object*/, p_vpi_delay /*delay_p*/) { _VL_VPI_UNI
 
 // value processing
 
-void vpi_get_value(vpiHandle object, p_vpi_value value_p) {
+template <typename T> void vpi_get_val(T* vop, p_vpi_value value_p) {
```

Comment:
Perhaps I'm misunderstanding something here, but the templated function was introduced to avoid duplicating 270 lines of code to support both ```VerilatedVpioVar``` and ```VerilatedVpioParam``` in ```vpi_get_value()```. 
The prototype ```vpi_get_value(vpiHandle object, p_vpi_value value_p)``` has remained the same and was moved down to line 1635. However, inside  ```vpi_get_value()```, we call the templated ```vpi_get_val<VerilatedVpioVar>()```  or ```vpi_get_val<VerilatedVpioParam>()``` depeding on the type of vpiHandle. The templated function is nearly the same code as what was in ```vpi_get_value()``` originally.
Please let me know if this is incorrect of if there is a better way to solve this issue.

Reply:
I'd suggest making a common function without a template, you can modify the two classes to inherit a base instead.  Also please rename the function to not match a VPI function (e.g. vl_vpi_...) as it's now internal-only.

Reply:
I've changed the function to take some pointers to relevant members which is common to both ```VerilatedVpioVar``` and ```VerilatedVpioParam```
```vl_get_value(const VerilatedVar* varp, void* varDatap, p_vpi_value valuep, const char* fullname)```